src/NuGet.Services.AzureSearch/SearchService/IndexOperation.cs

```diff
@@ -0,0 +1,45 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using Microsoft.Azure.Search.Models;
+
+namespace NuGet.Services.AzureSearch.SearchService
+{
+    public class IndexOperation
+    {
+        private IndexOperation(
+            IndexOperationType type,
+            string documentKey,
+            string searchText,
+            SearchParameters searchParameters)
+        {
+            Type = type;
+            DocumentKey = documentKey;
+            SearchText = searchText;
+            SearchParameters = searchParameters;
+        }
+
+        public IndexOperationType Type { get; }
+        public string DocumentKey { get; }
```

Comment:
>DocumentKey [](start = 22, length = 11)

Consider adding a comment like "null if Type is XYZ". Same comment for SearchText and SearchParameters

Reply:
Done.